src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Caching.cs

```diff
@@ -23,10 +23,55 @@ public sealed partial class JsonSerializerOptions
         // Simple LRU cache for the public (de)serialize entry points that avoid some lookups in _cachingContext.
         private volatile JsonTypeInfo? _lastTypeInfo;
 
+        /// <summary>
+        /// Gets the <see cref="JsonTypeInfo"/> contract metadata resolved by the current <see cref="JsonSerializerOptions"/> instance.
+        /// </summary>
+        /// <param name="type">The type to resolve contract metadata for.</param>
+        /// <returns>The contract metadata resolved for <paramref name="type"/>.</returns>
+        /// <remarks>
+        /// Returned metadata can be downcast to <see cref="JsonTypeInfo{T}"/> and used with the relevant <see cref="JsonSerializer"/> overloads.
+        ///
+        /// If the <see cref="JsonSerializerOptions"/> instance is locked for modification, the method will return a cached instance for the metadata.
+        /// </remarks>
+        [RequiresUnreferencedCode("Getting a metadata for a type may require reflection which depends on unreferenced code.")]
+        [RequiresDynamicCode("Getting a metadata for a type may require reflection which depends on runtime code generation.")]
+        public JsonTypeInfo GetTypeInfo(Type type)
```

Comment:
you might want to test scenario where this is called from resolver and causing some recursion to happen

Reply:
Ok

Reply:
Added, although it's simply validating that a resolver calling back into `options.GetTypeInfo` will prompt a stack overflow. This is fine, similar behaviors arise in improper uses of `option.GetConverter` as well.